module-2/backend/week_one.md

```diff
@@ -4,29 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+Model View Controller  
+5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?  
+This allows programmers to look at and use each others code with ease and understanding.   
+6. What types of variables are accessible in our view templates without explicitly passing them?  
+Instance variables
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
-  get '/horses' do
+  get '/horses' do  
+  
     erb :index
   end
-  ```
-
+  ```  
+@count = 1
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
-11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
+name = "Mr. Ed"
+erb :index, locals {name} (possibly something else with name?)
+9. What's the purpose of ERB?  
+Embeded Ruby - runs Ruby in an HTML template
+10. Why do I need a development AND test database?  
+To use a temporary database so the whole database doesn't have to run for each test and you can re-set the database between tests.  
+11. What's responsive design?  
+.. When CSS is designed so that it changes in relation to screen size.   
+12. What is CRUD and why is it important?  
+Create  
+Read  
+Update  
+Delete  
+This allows users full functionality with a database.  
+13. What does HTTP stand for?   
+Hyper Text Transfer Protocol  
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
+<%= %> Prints into the HTML  
+<% %> Runs Ruby code, does not print.  
+15. What's an ORM?  
+.. Object Relational Mapper
+16. What's the most commonly used ORM?  
+..  Active Record 
+17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.  
+..  C -GET new 
+      -POST submit/create new
+    R -GET index
+      -GET show 
+    U -GET edit page
+      -PUT submit
+    D -DELETE
+18. What's a migration?  
+.. Something that changes the structure of the database 
+19. When you create a migration, does it automatically modify your database?  
+..  First you have to create the migration then you have to run rake db:migrate
+20. How does a model relate to a database?   
+A model is a table in the database

```

Comment:
A model represents a table in the database, but I wouldn't consider them the same thing. You can create and run a migration to create a table in the database, and it will exist, but you need to create the model separately so that you can use it in your application (if you plan on using ActiveRecord and not simply making direct calls to the DB).